### PR TITLE
Prepare Identity June 2023 Release

### PR DESCRIPTION
**Pull request by billwert**

### CODE REVIEW

File: sdk/identity/azure-identity/CHANGELOG.md
Comment (vcolin7) [inline sdk/identity/azure-identity/CHANGELOG.md:7]: Has this change been reviewed by @srnagar or @JonathanGiles? At first glance it doesn't feel right to introduce a behavioral breaking change in a patch release.

**Review by vcolin7 — CHANGES_REQUESTED** [review]
Everything but the behavioral breaking change look good to me. Let's just get confirmation from an architect that they're ok with shipping it in this release :)

File: sdk/identity/azure-identity/CHANGELOG.md
Comment (billwert) [reply sdk/identity/azure-identity/CHANGELOG.md:7]: Yes. This shipped in 1.9. After we shipped a very late archboard decision came that changed the order, so we're catching up. @srnagar approved.

**Comment by billwert** [comment]
anomalydetector failures are unrelated.

**Review by g2vinay — APPROVED** [review]

**Review by vcolin7 — APPROVED** [review]
Approved after confirming architect approval.

**Pull request merged**
